### DISCUSSION TITLE
Details

### DISCUSSION THREAD

**Discussion by Vithanco** [hpcc-systems/hpcc-js-wasm#87] [Q&A]
I was reading through the documentation here. 
A few questions came up. Maybe you can answer them?

- How big is the wasm file(s) you produce?
- Why is the expat library included? 
- I assume that the WASM can run on any browser (according to WASM compliance) and has no further OS dependencies. Correct? (I got a bit insecure when reading build OS systems)

Many thanks in advance!
Klaus

**Comment by GordonSmith** [comment]
1. Currently the compressed download size is aprox. 388kb for graphviz.wasm and uncompressed size would be twice that.  
2. This repo is a placeholder for several wasm files (currently hosting 2).  I like to think of the wasm files the same way I do as any resource (image or font etc.), if your going to use one of the wasm files or wrappers then you don't need to publish it with your application, your end users will never know expat even existed.  
3. You are 100% correct - the build environment is somewhat complex compared to regular web applications, but this is something the end user is not concerned with.

**Reply by ghost** [reply]
Thanks for the help today (https://stackoverflow.com/questions/71571129/how-to-use-embedded-webassembly-in-vite)
I wonder whether you have as well a trick similar to `wasmFolder` on how to access a compressed wasm file?

**Reply by GordonSmith** [reply]
I am not sure I follow - in general the webserver and web client will compress the file by default?

**Reply by ghost** [reply]
of course! Thank you.

**Comment by Vithanco** [comment]
Thanks!

Do I understand this correctly? The Graphiz wasm  has no dependency on expat wasm? They just happen to be in the same repository?

cheers,
Klaus

**Reply by GordonSmith** [reply]
Yes that is correct - (but the c++ compilation of GraphViz does have a dependency on the c++ expat - for the "html node" support) and I had a need for a xml parser in JavaScript...

**Reply by Vithanco** [reply]
That is clear now. I like it - a lot! Good job!
Thank you!
